### DISCUSSION TITLE
WSDD Discovers client in NOTJOINED

### DISCUSSION THREAD

**Discussion by kmsmith222** [christgau/wsdd#123] [Q&A]
When I run wssd (installed with APT) thi found this in the logs: 
--

wsdd[21607]: 2021-11-12 18:05:28,195:wsdd INFO(pid 21607): (IP) .:3702(eth1) - - "ProbeMatches urn:uuid:0975e9a1-3cda-4663-ba2c-47d8327bd400 UDP" - -
Nov 12 18:05:28 server wsdd[21607]: 2021-11-12 18:05:28,210:wsdd INFO(pid 21607): discovered <clientname>  in NotJoined on x.x.x.x%

(Clientname and IP are correct for the machine)
--
I get this for 2 machines and neither are  accessible with either hostname or IP..
I opened up the ports on the firewall, and it did work perfectly for about a day. I updated python to 3.7 with no errors, removed the apt and github package as completely as I could and reinstalled it using the instructions here.  
It reports as Ubuntu 16.04 even though Ive done a dist upgrade. 
IF any light could be shed on this Id appreciate it.

**Comment by christgau** [comment]
> Nov 12 18:05:28 server wsdd[21607]: 2021-11-12 18:05:28,210:wsdd INFO(pid 21607): discovered in NotJoined on x.x.x.x%
> (Clientname and IP are correct for the machine)

So the discovery process is working correctly? "NotJoined" is an information that is provided by the discovered device. The device seems to provide an empty DisplayName property. That's why you only see 'discovered in'. The DisplayName would be seen between these two words if it was non-empty.

> I get this for 2 machines and neither are accessible with either hostname or IP..

I think there is a misunderstanding on what wsdd or the WSD protocol in general are doing here. With the discovery mode, other WSD compliant devices are searched and reported if they are found. This is what the protocol was designed for. The term "device" does not imply it is a Samba host. It can be a printer, scanner, (IP) camera or similar thing. Generally, you won't be able to "access" them via smb clients unless they run Samba, e.g. Note that wsdd does not check for the device type during discovery.

>  I opened up the ports on the firewall, and it did work perfectly for about a day. I updated python to 3.7 with no errors, removed the apt and github package as completely as I could and reinstalled it using the instructions here.

I do not see a Python or wsdd issue here. Connectivity issues are likely related to your network setup, but IMO this is not the best place to discuss these.

**Reply by kmsmith222** [reply]
I understand that WSD is for discovering and samba is for accessing. I have smbd running, but when I disable nmbd, 2 PCs disappear . I have opened the ports on the client and server's firewall per the instructions (3702, 5357 and relevant protocol), nothing in the logs dropped or rejected from those ports, even cleared the firewall to be sure that wasnt the issue.  I still get NOTJOINED. Also made sure SMB2 was running on both clients and smb.conf had the appropriate max/min protocol lines  
Is there anywhere else I might look to solve this? 
The current log is: 
sudo wsdd  -D -4 -i eth1 -v
2022-02-28 21:02:45,826:wsdd INFO(pid 166467): using pre-defined UUID d105aa46-ae54-57b0-993a-b63d543a1de6
2022-02-28 21:02:45,828:wsdd INFO(pid 166467): joined multicast group ('239.255.255.250', 3702) on 10.0.1.1%eth1
2022-02-28 21:02:45,829:wsdd INFO(pid 166467): scheduling Hello message via eth1 to ('239.255.255.250', 3702)
2022-02-28 21:02:46,832:wsdd INFO(pid 166467): scheduling Probe message via eth1 to ('239.255.255.250', 3702)
2022-02-28 21:02:46,838:wsdd INFO(pid 166467): 10.0.1.24:3702(eth1) - - "ProbeMatches urn:uuid:e6d074b0-9849-41e0-9308-6be2c6e683b6 UDP" - -
2022-02-28 21:02:46,857:wsdd INFO(pid 166467): discovered Lexmark MC3224adwe (LexiPrinter) on 10.0.1.24%eth1
2022-02-28 21:02:46,858:wsdd INFO(pid 166467): 10.0.1.31 - - "POST /d105aa46-ae54-57b0-993a-b63d543a1de6 HTTP/1.1" 200 -
2022-02-28 21:02:46,859:wsdd INFO(pid 166467): 10.0.1.31 - - "POST /d105aa46-ae54-57b0-993a-b63d543a1de6 HTTP/1.1" 200 -
2022-02-28 21:02:46,861:wsdd INFO(pid 166467): 10.0.1.6 - - "POST /d105aa46-ae54-57b0-993a-b63d543a1de6 HTTP/1.1" 200 -
2022-02-28 21:02:47,602:wsdd INFO(pid 166467): 10.0.1.6:3702(eth1) - - "ProbeMatches urn:uuid:e7a4bc9d-5d60-4460-aa65-d01e156647fd UDP" - -
2022-02-28 21:02:47,611:wsdd INFO(pid 166467): discovered A665 in NotJoined on 10.0.1.6%eth1
2022-02-28 21:02:48,764:wsdd INFO(pid 166467): 10.0.1.31:3702(eth1) - - "ProbeMatches urn:uuid:843ecf9c-e218-43c6-aae2-6dfb2884d988 UDP" - -
2022-02-28 21:02:48,770:wsdd INFO(pid 166467): discovered ASUSROG in NotJoined on 10.0.1.31%eth1
2022-02-28 21:03:02,627:wsdd INFO(pid 166467): 10.0.1.6:59124(eth1) - - "Resolve urn:uuid:7e3fe95f-7a0d-474e-9072-4274f9bf9f95 UDP" - -

**Reply by christgau** [reply]
> I understand that WSD is for discovering and samba is for accessing. 

Right.

Just to be clear: The intended use case is to run wsdd on a Linux machine that runs Samba, e.g. for file serving purposes. In this typical scenario, wsdd helps the _Windows_ clients (or more precisely, the user on the Windows machine) to find the Samba machine. Also, Samba enabled the access to the files in this typical scenario but actually the access is performed by the Windows clients. 

You can also use the Samba client to access Samba servers or Windows machines. 

Nevertheless, I am not sure if I understand your problem right. Could you please describe what you intend to achieve _and_ answer the questions below?! 

> I have smbd running, but when I disable nmbd, 2 PCs disappear . 

Where do the 2 PCs disappear? AFAIK, nmdb has nothing to do with wsdd.

> I have opened the ports on the client and server's firewall per the instructions (3702, 5357 and relevant protocol), nothing in the logs dropped or rejected from those ports, even cleared the firewall to be sure that wasnt the issue. I still get NOTJOINED. 

What do you expect to see instead of "NotJoined"?

> Also made sure SMB2 was running on both clients and smb.conf had the appropriate max/min protocol lines Is there anywhere else I might look to solve this? 

What are the clients you are referring to? Since you mention the smb.conf, I assume they are Linux machines

> The current log is: sudo wsdd -D -4 -i eth1 -v 

You are starting wsdd in order to discover other WSD compliant devices. This appears to succeed and wsdd prints the information it could get from the detected device. Note that the "NotJoined" is not something wsdd comes with, but it is an information provided by the device: 

https://github.com/christgau/wsdd/blob/c87819bda08acbe5913057770cc393c788577fb7/src/wsdd.py#L662-L664

Please also note that a WSD compliant device may not serve files. 

Besides that: You don't have to run wsdd under root privileges. This is a potential security risk.

**Comment by kmsmith222** [comment]
You can also use the Samba client to access Samba servers or Windows
machines.

Nevertheless, I am not sure if I understand your problem right. Could you
please describe what you intend to achieve *and* answer the questions below?

---

Just want all my devices to use WSD so I can stop using SMB1.


I have smbd running, but when I disable nmbd, 2 PCs disappear .

Where do the 2 PCs disappear? AFAIK, nmdb has nothing to do with wsdd.

---

I can sort the Win10 Network tab by discovery method.Two win10 clients are
detected using NETBIOS when the other win10 client and one Ubuntu server
are listed under WSD.

---

I have opened the ports on the client and server's firewall per the
instructions (3702, 5357 and relevant protocol), nothing in the logs
dropped or rejected from those ports, even cleared the firewall to be sure
that wasnt the issue. I still get NOTJOINED.

What do you expect to see instead of "NotJoined"?

--

Truthfully, Im not 100% sure as Im not that familiar with the WSD log
messages.  So the client is   "NOTJOINED"  to what?

---

Also made sure SMB2 was running on both clients and smb.conf had the
appropriate max/min protocol lines. Is there anywhere else I might look to
solve this?

What are the clients you are referring to? Since you mention the smb.conf,
I assume they are Linux machines

-- 

Sorry for not making that more clear. I have 3 Win10 clients and a printer
on an Ubuntu Server.

---

The current log is: sudo wsdd -D -4 -i eth1 -v--

---

You are starting wsdd in order to discover other WSD compliant devices.
This appears to succeed and wsdd prints the information it could get from
the detected device. Note that the "NotJoined" is not something wsdd comes
with, but it is an information provided by the device:

https://github.com/christgau/wsdd/blob/c87819bda08acbe5913057770cc393c788577fb7/src/wsdd.py#L662-L664

---

I have disabled SMB1 on the win10 devices in question, and enabled SMB2-3
via powershell and verified its active.
 ---

Please also note that a WSD compliant device may not serve files.

Thats why I kept SMBD running. I shut down NMBD for testing purposes.

--

Besides that: You don't have to run wsdd under root privileges. This is a
potential security risk.

Ok thanks, Ill keep that in mind.

----

I'm missing something, and its annoying me lol


Thanks,


Kevin

**Reply by christgau** [reply]
> Just want all my devices to use WSD so I can stop using SMB1. 
> [...]
>  I can sort the Win10 Network tab by discovery method.Two win10 clients are detected using NETBIOS when the other win10 client and one Ubuntu server are listed under WSD.

So you want the two Windows Client to be detected via WSD? If so, then this is not an issue related to wsdd. You have to change the configuration of Windows: The "Function Discovery Resource Publication" service needs to be started (and enabled) in order to announce a Windows host using the WSD protocol.


> > What do you expect to see instead of "NotJoined"?

> -- Truthfully, Im not 100% sure as Im not that familiar with the WSD log messages. So the client is "NOTJOINED" to what?

The devices are not joined to whatever the device consider to be joinable. That might be a AD domain or a Windows workgroup. But actually, I would not care about this. As I already noted, this information ("NotJoined") is provided by the device.
